zebra-chain/src/primitives/address.rs

```diff
@@ -0,0 +1,56 @@
+//! `zcash_address` conversion to `zebra_chain` address types.
+//!
+//! Usage: <https://docs.rs/zcash_address/0.2.0/zcash_address/trait.TryFromAddress.html#examples>
+
+use crate::{parameters::Network, transparent, BoxError};
+
+impl TryFrom<zcash_address::Network> for Network {
+    // TODO: better error type
+    type Error = BoxError;
+
+    fn try_from(network: zcash_address::Network) -> Result<Self, Self::Error> {
+        match network {
+            zcash_address::Network::Main => Ok(Network::Mainnet),
+            zcash_address::Network::Test => Ok(Network::Testnet),
+            zcash_address::Network::Regtest => Err("unsupported Zcash network parameters".into()),
+        }
+    }
+}
+
+impl From<Network> for zcash_address::Network {
+    fn from(network: Network) -> Self {
+        match network {
+            Network::Mainnet => zcash_address::Network::Main,
+            Network::Testnet => zcash_address::Network::Test,
+        }
+    }
+}
+
+// TODO:
+// addresses with embedded networks:
+//       implement TryFromRawAddress for sprout, sapling, unified/sapling, unified/orchard
+//       use convert_if_network(network.into()) to convert raw addresses
+// addresses which require a separate network:
+//       implement TryFromAddress for unified/transparent (p2pkh and p2sh)
```

Comment:
This might need some tweaks.

We don't have sapling or sprout address types, so we might need to re-export the `zcash_primitives` type. But we should get a `TryFromRawAddress` or `TryFromAddress` impl for free.

We do have an orchard type, so we can try to convert to that, if it contains enough information for the RPC. It's also ok to re-use the `zcash_primitives` type.

(And if you'd like, you can use the `zcash_primitives` transparent type as well.)